app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/BaseAppsmithRepositoryImpl.java

```diff
@@ -266,26 +268,35 @@ public Flux<T> queryAll(List<Criteria> criterias, AclPermission aclPermission, S
     }
 
     public Flux<T> queryAll(List<Criteria> criterias, List<String> includeFields, AclPermission aclPermission, Sort sort) {
+        return queryAll(criterias, includeFields, aclPermission, sort, NO_RECORD_LIMIT);
+    }
+
+    public Flux<T> queryAll(List<Criteria> criterias, List<String> includeFields, AclPermission aclPermission, Sort sort, int limit) {
         final ArrayList<Criteria> criteriaList = new ArrayList<>(criterias);
         return ReactiveSecurityContextHolder.getContext()
                 .map(ctx -> ctx.getAuthentication())
                 .map(auth -> auth.getPrincipal())
                 .flatMap(principal -> getAllPermissionGroupsForUser((User) principal))
-                .flatMapMany(permissionGroups -> queryAllWithPermissionGroups(criteriaList, includeFields, aclPermission, sort, permissionGroups));
+                .flatMapMany(permissionGroups -> queryAllWithPermissionGroups(criteriaList, includeFields, aclPermission, sort, permissionGroups, limit));
     }
 
     public Flux<T> queryAllWithPermissionGroups(List<Criteria> criterias,
-                                         List<String> includeFields,
-                                         AclPermission aclPermission,
-                                         Sort sort,
-                                         Set<String> permissionGroups) {
+                                                List<String> includeFields,
+                                                AclPermission aclPermission,
+                                                Sort sort,
+                                                Set<String> permissionGroups,
+                                                int limit) {
         final ArrayList<Criteria> criteriaList = new ArrayList<>(criterias);
         Query query = new Query();
         if (!CollectionUtils.isEmpty(includeFields)) {
             for (String includeField : includeFields) {
                 query.fields().include(includeField);
             }
         }
+
+        if (limit != -1) {
```

Comment:
Please use the introduced variable.
```suggestion
        if (limit != NO_RECORD_LIMIT) {
```